### PR TITLE
[feature/v12-onboarding] Unified Onboarding and Account Setup (Vanilla and Branded)

### PR DESCRIPTION
**Pull request by hosy**
## Description
Migrated the branded onboarding process into the vanilla add account view.
Adding accounts will be handled in the same view.

As an addition if no account was added, an onboarding view will be shown.

## Motivation and Context
Add accounts to the app and remove old branded onboarding.

## How Has This Been Tested?
- try vanilla onboarding
- try onboarding with customer file `Branding.plist`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] Added an issue with details about all relevant changes in the [**iOS documentation repository**](https://github.com/owncloud/docs-client-ios-app/issues).
- [x] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Added changelog files for the fixed issues in folder changelog/unreleased

### CODE REVIEW

File: ownCloud/App Controllers/AppRootViewController.swift [hunk 1]
```diff
@@ -116,7 +118,46 @@ open class AppRootViewController: EmbeddingViewController, BrowserNavigationView
...
+                messageView.cssSelector = .info
+                messageView.backgroundView?.cssSelector = .info
+                messageView.elementInsets = NSDirectionalEdgeInsets(top: 25, leading: 50, bottom: 50, trailing: 50)
+                
+                var addAccountTitle = "Add account".localized
+                if !VendorServices.shared.canAddAccount {
+                    addAccountTitle = "Login".localized
+                }
+                
+                messageView.elements?.append(.button(addAccountTitle, action: UIAction(handler: { [weak self] action in
```
Comment (felix-schwarz) [inline ownCloud/App Controllers/AppRootViewController.swift:137]: The buttons could also be included directly in the array passed to `.infoBox(additionalElements:)`.

File: ownCloudAppShared/UIKit Extension/UIImage+Extension.swift [hunk 2]
```diff
@@ -70,5 +70,22 @@ public extension UIImage {
 		self.draw(at: origin)
 
 		return UIGraphicsGetImageFromCurrentImageContext()?.withRenderingMode(self.renderingMode)
-	}
+    }
+    
+    public func withRoundedCorners(radius: CGFloat? = nil) -> UIImage? {
```
Comment (felix-schwarz) [inline ownCloudAppShared/UIKit Extension/UIImage+Extension.swift:75]: This method doesn't seem to be used anywhere.

**Review by felix-schwarz — CHANGES_REQUESTED** [review]

File: ownCloud/App Controllers/AppRootViewController.swift [hunk 3]
```diff
@@ -116,7 +118,46 @@ open class AppRootViewController: EmbeddingViewController, BrowserNavigationView
...
-		contentViewController = contentBrowserController
+		noBookmarkCondition = DataSourceCondition(.empty, with: OCBookmarkManager.shared.bookmarksDatasource, initial: true, action: { [weak self] condition in
+			if condition.fulfilled == true {
+				// No account available
+                let messageView = ComposedMessageView.infoBox(additionalElements: [
+                    .image(AccountSettingsProvider.shared.logo, size: CGSize(width: 128, height: 128)),
+                    .title(String(format: "Welcome to %@".localized, VendorServices.shared.appName), alignment: .centered, insets: NSDirectionalEdgeInsets(top: 25, leading: 0, bottom: 25, trailing: 0))
+                ])
+                messageView.cssSelector = .info
+                messageView.backgroundView?.cssSelector = .info
```
[the remark this answers is not in the document]
Comment (hosy) [reply ownCloud/App Controllers/AppRootViewController.swift]: Please can you show me an example of setting up a new css selector?

**Comment by hosy** [comment]
@felix-schwarz fixed CR findings and left a comment.

**Comment by felix-schwarz** [comment]
@hosy I added the CSS selector and added records for them in this commit:
https://github.com/owncloud/ios-app/pull/1208/commits/294ee5ce306f2cba4076289674fbff987d4438bc

**Review by felix-schwarz — APPROVED** [review]

**Comment by hosy** [comment]
@jesmrec do you want to have a look before merging?

**Comment by jesmrec** [comment]
i will, yes

**Comment by jesmrec** [comment]
about the first screen. Is that intended such black frame around the ownCloud blue? 

| iPhone | iPad |
| :-: | :-: |
|Screen Shot 2023-05-09 at 13 38 53|Screen Shot 2023-05-09 at 13 37 10|

**Review by jesmrec — APPROVED** [review]

**Comment by felix-schwarz** [comment]
Merged into https://github.com/owncloud/ios-app/pull/1095.

**Pull request merged**